File: responses/NKH74F-92U/longest_common_prefix.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(X, B):
    if X == 0 or X > len(B) or len(B) == 0:
        return 0
    else:
        max_product = 0
        for i in range(X-1, len(B)):
            product = 1
            for j in range(X):
                product *= B[i-j]
            if product > max_product:
                max_product = product
        return max_product

File: responses/A9KJ7K-GQ8/longest_common_prefix.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(A) {
    let done = false;
    let pos = 0;
    let prefix = "";
    let reducer = (acc, val) => {
        return acc && val.startsWith(prefix)
    }
    
    while (!done) {
        prefix += A[0][pos];
        if (A.reduce(reducer, prefix)) {
            pos += 1;   
        }
        else {
            prefix = prefix.substring(0, prefix.length - 1);
            done = true;
        }
    }
    return prefix;
}